components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2218,6 +2239,8 @@ where
             new_peer.peer.peer_stat = self.fsm.peer.peer_stat.clone();
             let campaigned = new_peer.peer.maybe_campaign(is_leader);
             new_peer.has_ready |= campaigned;
+            new_peer.peer.approximate_size = self.fsm.peer.approximate_size;
+            new_peer.peer.approximate_keys = self.fsm.peer.approximate_keys;
```

Comment:
We only need to change it when it is a leader. So you can do in in next statement.
```
if is_leader {
}
```

Reply:
Done.